File: examples-sdk/kotlin/020-basic-ai-ab-testing/main.kt
// 020-basic-ai-ab-testing — Kotlin SDK equivalent
// Compile: vil compile --from kotlin --input 020-basic-ai-ab-testing/main.kt --release

fun main() {
    val server = VilServer("ai-ab-testing-gateway", 8080)
    val ab = ServiceProcess("ab")
    ab.endpoint("POST", "/infer", "infer")
    ab.endpoint("GET", "/metrics", "metrics")
    ab.endpoint("POST", "/config", "update_config")
    server.service(ab)
    val root = ServiceProcess("root")
    root.endpoint("GET", "/", "index")
    server.service(root)
    server.compile()
}

File: examples-sdk/swift/020-basic-ai-ab-testing/main.swift
// 020-basic-ai-ab-testing — Swift SDK equivalent
// Compile: vil compile --from swift --input 020-basic-ai-ab-testing/main.swift --release

let server = VilServer(name: "ai-ab-testing-gateway", port: 8080)
let ab = ServiceProcess(name: "ab")
ab.endpoint(method: "POST", path: "/infer", handler: "infer")
ab.endpoint(method: "GET", path: "/metrics", handler: "metrics")
ab.endpoint(method: "POST", path: "/config", handler: "update_config")
server.service(ab)
let root = ServiceProcess(name: "root")
root.endpoint(method: "GET", path: "/", handler: "index")
server.service(root)
server.compile()
